### ISSUE TITLE
Acceptance test times out

### ISSUE THREAD

**Issue by knownasilya** [samselikoff/ember-cli-mirage#129]
```js
import Ember from 'ember';
import { module, test } from 'qunit';
import startApp from 'map/tests/helpers/start-app';
import PO from '../page-object';

const {
  fillable,
  visitable,
  text,
  clickable
} = PO;
var application;

module('Acceptance | submit feedback', {
  beforeEach: function() {
    application = startApp();
  },

  afterEach: function() {
    Ember.run(application, 'destroy');
  }
});

test('visiting /feedback', function(assert) {
  const feedback = PO.build({
    visit: visitable('/feedback'),
    closeModal: clickable('.modal .btn-primary'),

    form: {
      email: fillable('#feedback-email'),
      comment: fillable('#feedback-message'),
      submit: clickable('[type=submit]'),
      message: text('.ember-notify')
    }
  });


  feedback.visit()
    .closeModal();

  andThen(function () {
    assert.equal(currentURL(), '/feedback');
  });

  feedback.form()
    .email('test@blah.com')
    .comment('hello')
    .submit();

  andThen(function() {
    assert.equal(currentURL(), '');
    assert.equal(feedback.form().message(), 'Thank you for your feedback!');
  });
});
```

Which has mirage setup with the defaults,  namespace set to 'api' and renamed the contact fixture/factory to feedback. Only the post is used, `this.post('/feedbacks');`.

Everything works manually in development (not in test).

**Comment by seawatts** [comment]
Do you mean when you run `ember test` it builds and then just sits there and does not run any tests?

Because that's the issue I've been seeing ever since I upgraded to 0.0.29

**Comment by knownasilya** [comment]
I does run the tests, but gets stuck on the one acceptance test I have. Then it says that test timed out and continues to unit/jshint tests.

I even did `node debug ``which ember`` test` and that never even hit a debugger in the acceptance test.

**Comment by samselikoff** [comment]
try master

**Comment by blimmer** [comment]
I think this should be fixed, or there'd be more comments in here. Closing for now. Please reopen if there's an issue.

**Comment by richmolj** [comment]
I was getting this as well on a new project.

Turns out it's when your ajax call is being made but there is no  corresponding route in `mirage/config.js` (my issue was forgetting to prepend '/api/`). I think this issue should re-open to supply friendly error about what's going on.

**Comment by samselikoff** [comment]
you didn't see any error in your console, or the network tab?

**Comment by richmolj** [comment]
Didn't run in the browser but nothing in testem/phantom

**Comment by g-cassie** [comment]
I am experiencing this issue.  Running the tests in the browser and I do not get any console log statements from mirage about what api routes are being hit and what the responses are.  I am sure the problem is in my test code but it makes it difficult to debug without those log statements to see what mirage is responding to successfully/unsuccessfully.

**Comment by g-cassie** [comment]
I found the switch to turn logging in tests on now so that helps a lot.  Unfortunately now I see that all my api request are being responded to and the tests are still hanging until they time out...

**Comment by samselikoff** [comment]
@g-cassie try `server.logging = true` in your tests to diagnose

**Comment by samselikoff** [comment]
Could I see your test?

**Comment by g-cassie** [comment]
I solved it... turned out to be a totally unrelated to ember-cli-mirage and just fluke that I found this issue.  I had a function calling itself in Ember.run.next() which was creating an infinite run loop.  Sorry for the noise here!

**Comment by samselikoff** [comment]
I believe the error message now supplied addresses this issue. Please lmk if anyone still needs help!

**Comment by visualjeff** [comment]
I hate to says this, but I'm getting a "test timed out" when I attempt to run my acceptance test.
I used pretender by itself in the past and had not problem with acceptance testing but I can't seem to make mirage work.

**My current environment is:**

  version: 1.13.15
  node: 4.2.4
  npm: 2.14.10
  os: linux x64

**Here is the actual output:**

07:30:57 {master} ~/WebstormProjects/bact-web/bact-client$ ember t
version: 1.13.15

Built project successfully. Stored in "/home/jeffg/WebstormProjects/bact-web/bact-client/tmp/class-tests_dist-EIvjEuIy.tmp".
not ok 1 PhantomJS 1.9 - Acceptance | locations: visiting /locations
    ---
        actual: >
            null
        stack: >
                at http://localhost:7357/assets/test-support.js:2396
        message: >
           ** Test timed out**
        Log: |


**Test contains the following source:**

import Ember from 'ember';
import { module, test } from 'qunit';
import startApp from '../helpers/start-app';

module('Acceptance | locations', {
    beforeEach() {
        this.application = startApp();
    },

    afterEach() {
        Ember.run(this.application, 'destroy');
    }
});


test('visiting /locations', function(assert) {
  server.logging = true;
  server.createList('location', 5);
  
  visit(/locations');

  andThen(function() {
    assert.equal(currentURL(), '/locations');
  });
});

Not much in the way of an error message from the browser console....  So I'm looking for a pointer.  Any ideas?  The browser console barfs a suggestion that maybe I haven't included the namespace info.  But when I add in namespace info it makes little difference.

Any suggestions?

**Comment by g-cassie** [comment]
@visualjeff I would look over any use of `Ember.run.*` in your application code - any kind of timer could cause a timeout (whenever you use timers you should set them in a config variable that you then override to zero during your tests).  In my case it was `Ember.run.next()` that was creating an infinite loop in testing only due to some weird technical difference between the test and dev environments.

**Comment by visualjeff** [comment]
Thanks.  I will do.  I suspect you may be right.

**Comment by visualjeff** [comment]
So I greped my projects app directory and found one use of Ember.run.  It pertained to foundationjs.  

So I went from...
```
import Ember from 'ember';

export default Ember.Component.extend({
  didInsertElement: function() {
      this._super();
      **Ember.run**.scheduleOnce('afterRender', this, function() {
          Ember.$(document).foundation();
      });
  }
});
```

To...
```
import Ember from 'ember';

export default Ember.Component.extend({
  didInsertElement: function() {
      this._super();
      Ember.$(document).foundation();
  }
});
```

So I removed the Ember.run.  But I still see a timeout out in the acceptance tests.

**Comment by g-cassie** [comment]
Have you isolated this issue to a single test or does it happen on every single acceptance test?

**Comment by g-cassie** [comment]
Another thing you can do is turn on logging for mirage:

```
module('Acceptance | My Module', {
  setup: function() {
    application = startApp();
    server.logging = true;
  },

  teardown: function() {
    Ember.run(application, 'destroy');
  }
});
```

It could be that a promise from ember-data is never resolving because the response from mirage is incorrect.  

Are you able to use your app in development mode with mirage providing the backend?

**Comment by visualjeff** [comment]
Here are two versions of the same test.  Both time out.

Version 1:
```
import { test } from 'qunit';
import moduleForAcceptance from '../helpers/module-for-acceptance';

moduleForAcceptance('Acceptance | locations');

test('visiting /locations', function(assert) {
  server.logging = true;
  server.createList('location', 5);
  
  visit('/');

  andThen(function() {
    assert.equal(currentURL(), '/locations');
  });
});
```

Version 2:
```
import Ember from 'ember';
import { module, test } from 'qunit';
import startApp from '../helpers/start-app';

module('Acceptance | My Module', {
  setup: function() {
    this.application = startApp();
    server.logging = true;
  },

  teardown: function() {
    Ember.run(this.application, 'destroy');
  }
});


test('visiting /locations', function(assert) {
  server.createList('location', 5);
  
  visit('/');

  andThen(function() {
    assert.equal(currentURL(), '/locations');
  });
});
```

The console states it can't find the api.  Suggest adding that I'm missing namespace info.  The does have a namespace of api/v1.  When I step though the app manually everything works fine.   So why would I need to a namespace info to a test if I'm visiting the default / url of the site?  I visit / then I'm transitioned to /locations.  Isn't it the adapter uses the namespace when invoking the supporting service...

**Comment by g-cassie** [comment]
Do you see logs in the console showing that the requests are being successfully made and received by mirage?  Should look something like this:

![screen shot 2016-03-07 at 3 17 37 pm](https://cloud.githubusercontent.com/assets/4053345/13582046/f6ad1c06-e477-11e5-8070-efbdca75adc4.png)

This debugging is kind of getting outside the scope of GH but if you want to message me on the ember slack I can try to suggest some other ideas.

**Comment by visualjeff** [comment]
Figured it out.  Thank you for being soooo patient.   Maybe the error message could suggest "if you're app is using a namespace within your API (like api or api/v1), double check the namespace setting you have defined for your test environment (located in config/environment.js file)".  Maybe that's too wordy?

**Comment by samselikoff** [comment]
@visualjeff sorry you ran into this, glad you figured it out! 

I'm curious, why do you have a different namespace defined for your test environment? That seems like it could be a smell to me.

**Comment by bdiz** [comment]
@visualjeff I'm getting same error you were getting. Can you post the config/environment.js namespace setting you are referring to? I can't find in ember documentation anything about setting namespace in this file so I'm confused. Thanks.

**Comment by samselikoff** [comment]
Typically `namespace` is defined on the Application Adapter (assuming you're using Ember Data), so I'm not sure which property @visualjeff is referring to.

**Comment by PratheepV** [comment]
Is there any way to debug this issue. I am getting it only in my CI. Local its working fine.

**Comment by rafael-paiva** [comment]
I'm having the same issue described above. My acceptance test pass locally but fail on my CI which makes harder to investigate :(

**Comment by visualjeff** [comment]
Grep your project and mirage's config files for the property "namespace".    These values need to match.   Side note:  I use [the silver searcher](https://github.com/ggreer/the_silver_searcher) for a really fast grep.
